### DISCUSSION TITLE
Typescript errors in node_modules

### DISCUSSION THREAD

**Discussion by ericlifs** [mmazzarolo/react-native-universal-monorepo#32] [Q&A]
Hey everyone! I'm porting my current monorepo to this awesome setup and I'm facing some new issues that I didn't have before. I have some 3rd party libraries installed in my repo and I'm getting some errors on those libraries' code that I didn't have before.

The thing is that even though I have my `node_modules` folder excluded from transpiling within my tsconfig.json file (the same way you did), I'm still getting typescript errors from those libraries.

You can replicate this issue just by cloning your repo and installing `react-native-linear-gradient` or `react-native-dialog` in the app package, add some usage in the App.tsx file and then start the web package. You will see something like:

<img width="463" alt="image" src="https://user-images.githubusercontent.com/7376892/141188622-55da3cb6-c7f6-47b3-856e-77bec991a78d.png">

<img width="504" alt="image" src="https://user-images.githubusercontent.com/7376892/141130753-e06ed09b-7810-44fd-8c37-07beb20a5e4c.png">

<img width="512" alt="image" src="https://user-images.githubusercontent.com/7376892/141130350-313d16bc-deb8-4e2d-8f62-8ed9bec55379.png">

<img width="376" alt="image" src="https://user-images.githubusercontent.com/7376892/141130890-b8831e8c-9daa-42f1-b52b-905ba0c923d5.png">

<img width="336" alt="image" src="https://user-images.githubusercontent.com/7376892/141131093-f60ccf58-2fb0-4c24-8426-2c3002bbc885.png">

I don't know if I have to configure babel in a certain way in order to make this error disappear, but my previous monorepo was also using a CRA scaffold for the web package and I was not getting any of this errors.

Thank you very much,

Eric.

**Comment by ericlifs** [comment]
I have just tested this again starting from scratch on your monorepo scaffold and I'm seeing the same issues. For example if you want to install `react-native-linear-gradient` for the web you need to:
- Run `yarn add react-native-linear-gradient` on the app package
- Run `yarn add react-native-web-linear-gradient` on the app package
- Add the webpack's alias section to the web's `craco.config.js`

```
webpack: {
    alias: {
      'react-native-linear-gradient': 'react-native-web-linear-gradient',
    },
    ...
}
```

- Use it anywhere in app's App.tsx:

```
<LinearGradient colors={['#4c669f', '#3b5998', '#192f6a']} style={styles.linearGradient}>
  <Text style={styles.buttonText}>
    Sign in with Facebook
  </Text>
</LinearGradient>
```

- And add the styles

```
linearGradient: {
    flex: 1,
    paddingLeft: 15,
    paddingRight: 15,
    borderRadius: 5
  },
  buttonText: {
    fontSize: 18,
    fontFamily: 'Gill Sans',
    textAlign: 'center',
    margin: 10,
    color: '#ffffff',
    backgroundColor: 'transparent',
  },
```

- Start the web app and you will get those errors:

<img width="348" alt="image" src="https://user-images.githubusercontent.com/7376892/141474947-79d0990d-aa15-4f66-b1cf-ea49baf1cceb.png">

I'm not sure if this is something that has happened in the past.

Thanks,

Eric.

**Comment by mmazzarolo** [comment]
@ericlifs sorry for being late, I'm busy these days. 
Can you try following the same flow we're using with the `@react-native-async-storage/async-storage` package? It's a native module as well, so I would expect `react-native-linear-gradient` to work in the same way.
